### PR TITLE
[BD-6] Add python 3.8 tests

### PR DESCRIPTION
**Pull request by morenol**
This PR:
 * Add tests in python3.8
 * Separate doc tests in their own travis worker
 * Upgrade dependencies
 * Quality fixes for new pylint version

The failure of tests is a known issue: See https://openedx.atlassian.net/browse/ARCHBOM-1061
I separated the problematic tests in their own worker (doc tests)

## Reviewers
- [ ] @ericfab179
- [x] @awais786 
- [x] Ready for edX review

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-4461](https://openedx.atlassian.net/browse/OSPR-4461) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by mraarif** [comment]
build for doc is failing under both python versions, maybe that needs to be fixed and commits should be squashed.

**Comment by morenol** [comment]
> build for doc is failing under both python versions, maybe that needs to be fixed and commits should be squashed.

Yes, I am going to squash them. The explanation of the failure is described in the link that I shared in the description of the PR. It is not clear to my which of these options I should take to solve the problem. And also I think that I can fix that in other PR.

**Comment by morenol** [comment]
@mraarif I squashed and then added a commit that solves the test, but that may fail in the future if github decides to change their search page again

**Comment by mraarif** [comment]
It should be fine for now I think if the tests fail and get reported I believe the team owning this will take care of this.

**Comment by morenol** [comment]
Perfect, @awais786 please review and merge if you think that this is ready.

File: docs/code/round_3/pages.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def search_results(self):
         """
         Return a list of results returned from a search
         """
-        return self.q(css='ul.repo-list > li > div > h3 > a').text
+        return self.q(css='ul.repo-list > li > div > div > a').text
```
Comment (awais786) [inline docs/code/round_3/pages.py:24]: @morenol whats a purpose of this change?

Comment (awais786) [reply docs/code/round_3/pages.py:24]: Ok fixing a test.

Comment (morenol) [reply docs/code/round_3/pages.py:24]: To make the tests pass, Github changed the style of that page that is used in the test so I had to change that.

**Comment by edx-webhook** [comment]
@morenol 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/bok-choy&entry.867055334=https://github.com/edx/bok-choy/pull/263&entry.1484655318=https://github.com/morenol&entry.752974735=2020-05-04+16:37&entry.1917517419=2020-05-08+17:07&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**